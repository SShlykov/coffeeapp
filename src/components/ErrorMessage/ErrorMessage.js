import React from "react";
import {Link} from "react-router-dom";

const ErrorMessage = ({error, button}) => {
    let errorMes= `Error status: ${error}`;
    switch (error) {
        case "404":
            errorMes = `data not found`;
            break;
        case "408":
            errorMes = `request timeout`;
            break;
        case "410":
            errorMes = `gone`;
            break;
        default:
            errorMes= `Error status: ${error}`;
    }
    const link = (button)? <Link to="/" className="preview__btn black-btn">back to main</Link>:null;
    return (
        <div className="error">
            <h2 className="error-title center">{error}</h2>
            <div className="center">{errorMes}</div>
            <div className="center error-description">We are about to fix it</div>
            {link}
        </div>
    )
};
export default ErrorMessage;