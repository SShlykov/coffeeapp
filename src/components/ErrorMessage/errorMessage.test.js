import React from 'react';
import ErrorMessage from "./ErrorMessage";
import {shallow} from 'enzyme';

describe('Testing <ErrorMessage/>', () => {
    const error = shallow(<ErrorMessage/>);
    describe('Testing snap & state', () => {
        it('ErrorMessage bar have rendered correctly', () => {
            expect(error).toMatchSnapshot();
        });    
    });
});