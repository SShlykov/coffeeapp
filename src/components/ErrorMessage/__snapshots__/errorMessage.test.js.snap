// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ErrorMessage/> Testing snap & state ErrorMessage bar have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorMessage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="error-title center"
        />,
        <div
          className="center"
        >
          Error status: undefined
        </div>,
        <div
          className="center error-description"
        >
          We are about to fix it
        </div>,
        null,
      ],
      "className": "error",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "error-title center",
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Error status: undefined",
          "className": "center",
        },
        "ref": null,
        "rendered": "Error status: undefined",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "We are about to fix it",
          "className": "center error-description",
        },
        "ref": null,
        "rendered": "We are about to fix it",
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="error-title center"
          />,
          <div
            className="center"
          >
            Error status: undefined
          </div>,
          <div
            className="center error-description"
          >
            We are about to fix it
          </div>,
          null,
        ],
        "className": "error",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "error-title center",
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Error status: undefined",
            "className": "center",
          },
          "ref": null,
          "rendered": "Error status: undefined",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "We are about to fix it",
            "className": "center error-description",
          },
          "ref": null,
          "rendered": "We are about to fix it",
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
